lib/go-backoff/backoff.go

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package backoff
+
+/*
+ * Used to return an exponentially increasing time.Duration value that may be used
```

Comment:
The package-level comment is great. But, it needs to be immediately before the package declaration, and start with `Package backoff ...` to work with GoDoc. Also, the intermediary C-style `*` will break GoDoc I believe. See https://blog.golang.org/godoc-documenting-go-code 

```go
/*
Package backoff is used to return an exponentially increasing time.Duration value that may be used to sleep between retries.
 */
package backoff
```

Alternatively, if you put this in the `go-util` package, this should be a symbol-level comment on `type BackOff interface` or `type BackOff struct`, `//Backoff is used to ...`